src/Common.Headers/StringUtils.h

```diff
@@ -77,5 +79,9 @@ class StringUtils
         }
 
         return S_OK;
+#else
+        LPWSTR pwszMore2 = const_cast<LPWSTR>(pwszMore);
+        return PathCchAppend(pwszPath, MAX_PATH, pwszMore2);
```

Comment:
Isn't this defined as:

```
HRESULT PathCchAppend(
  PWSTR  pszPath,
  size_t cchPath,
  PCWSTR pszMore
);
```

so LPCWSTR should work for pszMore. Also, I think a lot of the code before the Win7 support change passed in all kinds of variations of `const WCHAR*`.

Reply:
It's defined in the palrt2.h as:

```
extern "C" HRESULT PathCchAppend(
    _Inout_  WCHAR*  wszPath,
    _In_     size_t cchPath,
    _In_opt_ WCHAR* wszMore
);
```

Should I update this instead?

Reply:
We should check what the real signature was when the pal was built from dotnet/coreclr.

Reply:
The declaration is here: https://github.com/microsoft/CLRInstrumentationEngine/blob/43c4e97b8dbde7d8b2ca1a5e94c40ba8a53f5a22/src/unix/inc/palrt2.h#L646
The implementation is here: https://github.com/microsoft/CLRInstrumentationEngine/blob/43c4e97b8dbde7d8b2ca1a5e94c40ba8a53f5a22/src/unix/src/misc/file.cpp#L73

So this function is completely within out control.

I would say revert the header change that you made and just try to compile without having to do the const_cast. It was working before the Win 7 support change and it should work still (I don't see anything obvious that suggests it shouldn't work given how it worked before the Win 7 support change).

If it doesn't compile, then change both the declaration and implementation to match the signature that was defined for Windows:
```
HRESULT
PathCchAppend(
    _Inout_updates_(cchPath) PWSTR pszPath,
    _In_ size_t cchPath,
    _In_opt_ PCWSTR pszMore
    );
```

so that it same for Windows and non-Windows.